pkg/controllers/provisioning/v1alpha1/reallocation/controller.go

```diff
@@ -62,19 +60,37 @@ func (c *Controller) Name() string {
 func NewController(kubeClient client.Client, coreV1Client corev1.CoreV1Interface, cloudProvider cloudprovider.Factory) *Controller {
 	return &Controller{
 		utilization:   &Utilization{kubeClient: kubeClient},
-		terminator:    &Terminator{kubeClient: kubeClient, cloudprovider: cloudProvider, coreV1Client: coreV1Client},
 		cloudProvider: cloudProvider,
 	}
 }
 
 // Reconcile executes a reallocation control loop for the resource
-func (c *Controller) Reconcile(ctx context.Context, object controllers.Object) (reconcile.Result, error) {
+func (c *Controller) Reconcile(ctx context.Context, object client.Object) (reconcile.Result, error) {
 	provisioner := object.(*v1alpha1.Provisioner)
-	if err := c.utilization.Reconcile(ctx, provisioner); err != nil {
-		return reconcile.Result{}, fmt.Errorf("reconciling utilization sub-controller, %w", err)
+
+	// Mark ProvisionerStatus if Reconcile fails
+	var err error
```

Comment:
I think we should remove this in this controller. Right now there's actually a race condition. If allocator writes true and reallocator writes false, it will flip flop.

Reply:
I changed it so that the Provisioner would be set to StatusCondition true while we figure out what to do with this race condition. I'll cut an issue as well.